apps/assistants/sync.py

```diff
@@ -18,38 +47,118 @@ def push_assistant_to_openai(assistant: OpenAiAssistant):
         assistant.save()
 
 
+@wrap_openai_errors
+def push_file_to_openai(assistant: OpenAiAssistant, file: File):
+    client = assistant.llm_provider.get_llm_service().get_raw_client()
+    with file.file.open("rb") as fh:
+        bytesio = BytesIO(fh.read())
+    openai_file = client.files.create(
+        file=(file.name, bytesio),
+        purpose="assistants",
+    )
+    file.external_id = openai_file.id
+    file.external_source = "openai"
+    file.save()
+
+
+@wrap_openai_errors
+def delete_file_from_openai(assistant: OpenAiAssistant, file: File):
+    if not file.external_id or file.external_source != "openai":
```

Comment:
I'm just trying to wrap my head around the usage of `external_source`. Looks like it's only ever used for this check here. This makes me wonder if the `external_id` would not be enough safeguard (assuming its some sort of uuid that is)? Unless we want to cater for cases where the external id is a plain integer accors multiple external sources, in which case one can accidentely remove a file from the wrong external source.

Is this the intented usage of the `external_source` field?

Reply:
This is designed to cater for multiple places where files may be pushed to in future e.g. if we use an external vector DB or push files to S3 or something else. It's a bit of over engineering at this point.